text/0000-opencensus-migration-plan.md

```diff
@@ -0,0 +1,141 @@
+# OpenCensus To OpenTelemetry Migration Plan
+
+Define the process of migrating applications, libraries, and agents from [OpenCensus](https://opencensus.io/) to
+OpenTelemetry.
+
+## Motivation
+
+While OpenTelemetry has a specification for an
+[OpenCensus trace bridge](https://github.com/open-telemetry/opentelemetry-specification/blob/87a5ed7f0d4c403e2b336f275ce3e7fd66a8041b/specification/compatibility/opencensus.md),
+it lacks a broader context on how such bridges should be used to perform a migration from
+OpenCensus to OpenTelemetry. Additionally, OpenTelemetry should provide guidance to libraries
+currently using OpenCensus on how to migrate instrumentation to OpenTelemetry.
+
+## Explanation
+
+Migrating from OpenCensus to OpenTelemetry may require breaking changes to the telemetry produced
+because of:
+
+* Differences in semantic conventions for names and attributes
+* Data model differences
+* Instrumentation API feature differences
+* Differences between equivalent OC and OTel exporters
+
+Requirements:
+
+* Give application developers control over _when_ breaking changes are introduced.
+* No "long tail" of breaking changes to users when migrating.
+* Make it possible to migrate instrumentation without breaking the resulting telemetry.
+* Library maintainers should not need to maintain OpenCensus and OpenTelemetry instrumentation in parallel.
+
+This proposed process groups most breaking changes into the installation of the bridge. This gives
+users control over introducing the initial breaking change, and makes subsequent migrations of
+instrumentation (including in third-party libraries) non-breaking.
+
+## Migration Plans
+
+### Migrating from the OC Agent and Protocol
+
+Starting with a deployment of the OC agent, using the OC protocol, migrate by:
+
+1. Deploy the OpenTelemetry collector with OpenCensus and OTLP receivers and equivalent processors and exporters.
+2. **breaking**: For each workload sending the OC protocol, change to sending to the OpenTelemetry collector's OpenCensus receiver.
```

Comment:
why is this a breaking change? The app still sends the same data. Is OTEL's OC receiver going to rewrite the data, e.g. rename the attributes?

Reply:
Since I haven't defined what constitutes a breaking change, i've taken a fairly strict definition. The OTel collector's OpenCensus receiver translates OpenCensus metrics + spans to OTLP internally, which is a lossy translation.  For example, child span count for spans and GaugeDistribution-typed or the sum of squared deviation for metrics are lost when translating from OpenCensus to OTLP.  An OpenCensus exporter might've been able to use those fields, but the OpenTelemetry version of the exporter can't.